packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ConstructorProvider.cs

```diff
@@ -50,5 +50,17 @@ public ConstructorProvider(ConstructorSignature signature, ValueExpression bodyE
                 ? MethodProviderHelpers.BuildXmlDocs(signature.Parameters, signature.Description, null, null)
                 : null);
         }
+
+        /// <summary>
+        /// Builds a constructor for the enclosing type for mocking purposes.
+        /// </summary>
+        /// <param name="enclosingType">The type provider to build the constructor for.</param>
+        public static ConstructorProvider BuildMockingConstructor(TypeProvider enclosingType)
```

Comment:
lets start internal for now

Reply:
It needs to be public if we want to consume in SCM plugin's clientprovider.

Reply:
For small helpers like this I am ok with shared source (only because the two libraries ship together), but we shouldn't have a public API.